fjord/base/utils.py

```diff
@@ -387,13 +403,15 @@ def parse_rate(self, rate):
         return (num, (multiplier * period))
 
     def allow_request(self, request, view):
+        already_limited = getattr(request, 'limited', False)
         ratelimited = is_ratelimited(
-            request=request, increment=True, ip=False, method=self.methods,
-            field=None, rate=self.rate, keys=self.keyfun)
+            request=request, group=self.rulename,
+            key=self.keyfun, rate=self.rate, method=self.methods,
+            increment=True)
 
         if ratelimited:
-            # Failed rate-limiting, so this request is not allowed.
-            statsd.incr('throttled.' + self.rulename)
+            if not already_limited:
+                statsd.incr('throttled.' + self.rulename)
             return self.throttle_failure()
```

Comment:
I added the `already_limited` thing because otherwise in theory a request for a view with multiple throttles that all fail will do multiple `statsd.incr` calls. In practice, it might be moot since django rest framework might halt things at the first throttle that fails. I don't really know and don't know if it'll change at some point in the future.